MetaMask/metamask-extension
# fix: Error prompt when canceling a transaction

## **Description**
Error is prompted in the console when attempting to cancel a transaction and transaction is not cancelled.

The error comes from a call to selector where we are passing in `state.metamask` object instead of state object. To fix, we pass in an object where `metamask` is a property and value is `state.metamask`

## **Expected behavior**

No errors are shown and transaction is cancelled.

## **Manual testing steps**

1. Open MetaMask and enter your password
2. Switch to a test network: e.g. Goerli
3. Proceed to the send flow
4. Send 0.1 ETH to an address: e.g. 0x3D0b9b4056D86f8d9889B2C8e26FFCedC62D3036
5. Edit the gas settings in Market --> Advanced: Max base fee to 0.0000001 and Priority Fee to 0.0000001
6. Confirm the transaction
7. Cancel transaction

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/120041701/d371c902-b0bb-4e52-b152-549f1a487375

### **After**

https://github.com/MetaMask/metamask-extension/assets/44811/40dd2071-a9ca-4d31-9241-dfc76aaacd9f


## **Related issues**

_Fixes #21261

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
